services/recovery_method.js

```diff
@@ -1,69 +1,215 @@
-    deleteOtherRecoveryMethods({ accountId, detail }) {
-        return SequelizeRecoveryMethods.deleteOtherRecoveryMethods({ accountId, detail });
-    },
+        deleteOtherRecoveryMethods({ accountId, detail }) {
+            if (!USE_DYNAMODB) {
+                return this.sequelize.deleteOtherRecoveryMethods({ accountId, detail });
+            }
+            return this.db.listRecoveryMethodsByAccountId(accountId)
+                .filter((recoveryMethod) => recoveryMethod.detail !== detail)
+                .map((recoveryMethod) => deleteRecoveryMethod(recoveryMethod));
+        },
 
-    deleteRecoveryMethod({ accountId, kind, publicKey }) {
-        return SequelizeRecoveryMethods.deleteRecoveryMethod({ accountId, kind, publicKey });
-    },
+        deleteRecoveryMethod({ accountId, kind, publicKey }) {
+            if (!USE_DYNAMODB) {
+                return this.sequelize.deleteRecoveryMethod({ accountId, kind, publicKey });
+            }
+            return this.db.listRecoveryMethodsByAccountId(accountId)
+                .filter((recoveryMethod) => recoveryMethod.kind === kind && recoveryMethod.publicKey === publicKey)
+                .map((recoveryMethod) => deleteRecoveryMethod(recoveryMethod));
+        },
 
-    getTwoFactorRecoveryMethod(accountId) {
-        return SequelizeRecoveryMethods.getTwoFactorRecoveryMethod(accountId);
-    },
+        getTwoFactorRecoveryMethod(accountId) {
+            if (!USE_DYNAMODB) {
+                return this.sequelize.getTwoFactorRecoveryMethod(accountId);
+            }
+            return this.db.listRecoveryMethodsByAccountId(accountId)
+                .filter((recoveryMethod) => recoveryMethod.kind.startsWith('2fa-'))
+                .get(0)
+                .then((method) => method || null);
+        },
 
-    isTwoFactorRequestExpired({ updatedAt }) {
-        return updatedAt < Date.now() - TWO_FACTOR_REQUEST_DURATION_MS;
-    },
+        isTwoFactorRequestExpired({ updatedAt }) {
+            return updatedAt < (Date.now() - TWO_FACTOR_REQUEST_DURATION_MS);
+        },
 
-    listAllRecoveryMethods(accountId) {
-        return SequelizeRecoveryMethods.listAllRecoveryMethods(accountId);
-    },
+        listAllRecoveryMethods(accountId) {
+            if (!USE_DYNAMODB) {
+                return this.sequelize.listAllRecoveryMethods(accountId);
+            }
+            return this.db.listRecoveryMethodsByAccountId(accountId)
+                .map(({ securityCode, ...recoveryMethod }) => ({
+                    ...recoveryMethod,
+                    confirmed: !securityCode,
+                    requestId: parseInt(recoveryMethod.requestId, 10),
+                }));
+        },
 
-    listRecoveryMethods({ accountId, detail, kind, publicKey, securityCode }) {
-        return SequelizeRecoveryMethods.listRecoveryMethods({ accountId, detail, kind, publicKey, securityCode });
-    },
+        async listRecoveryMethods({ accountId, detail, kind, publicKey, securityCode }) {
```

Comment:
Looks like the DynamoDB portion of this method only ever returns an array of length 0 or 1, is that correct? Is it only possible to have only one recovery method? Why do we need the array?